File: Python/5kyu-Pete-the-baker.test.py

import pytest

test.describe('Testing Pete, the Baker')
test.it('gives us the right number of cakes')

recipe = {"flour": 500, "sugar": 200, "eggs": 1}
available = {"flour": 1200, "sugar": 1200, "eggs": 5, "milk": 200}
test.assert_equals(cakes(recipe, available), 2, 'Wrong result for example #1')

recipe = {"apples": 3, "flour": 300, "sugar": 150, "milk": 100, "oil": 100}
available = {"sugar": 500, "flour": 2000, "milk": 2000}
test.assert_equals(cakes(recipe, available), 0, 'Wrong result for example #2')

File: JS/tests/5kyu--Pete-the-baker.test.js
const { cakes } = require("../5kyu--Pete-the-baker.js")

describe('""Pete, the baker" tests:', function () {
  test("#1", () => {
    recipe = { flour: 500, sugar: 200, eggs: 1 }
    available = { flour: 1200, sugar: 1200, eggs: 5, milk: 200 }
    expect(cakes(recipe, available)).toEqual(2)
  })
  test("#2", () => {
    recipe = { apples: 3, flour: 300, sugar: 150, milk: 100, oil: 100 }
    available = { sugar: 500, flour: 2000, milk: 2000 }
    expect(cakes(recipe, available)).toEqual(0)
  })
})
